execution_engine/src/shared/gas.rs

```diff
@@ -40,170 +42,195 @@ impl fmt::Display for Gas {
     }
 }
 
-impl std::ops::Add for Gas {
-    type Output = Gas;
+impl CheckedAdd for Gas {
+    fn checked_add(&self, v: &Self) -> Option<Self> {
+        self.0.checked_add(v.0).map(Gas)
+    }
+}
 
-    fn add(self, rhs: Self) -> Self::Output {
-        let val = self.value() + rhs.value();
-        Gas::new(val)
+impl CheckedMul for Gas {
+    fn checked_mul(&self, v: &Self) -> Option<Self> {
+        self.0.checked_mul(v.0).map(Gas)
     }
 }
 
-impl std::ops::Sub for Gas {
-    type Output = Gas;
+impl CheckedSub for Gas {
+    fn checked_sub(&self, v: &Self) -> Option<Self> {
+        self.0.checked_sub(v.0).map(Gas)
+    }
+}
 
-    fn sub(self, rhs: Self) -> Self::Output {
-        let val = self.value() - rhs.value();
-        Gas::new(val)
+impl Sum for Gas {
```

Comment:
As I [mention above][1], I don't understand why we would ever want to add gas without checking for overflow...

[1]: https://github.com/casper-network/casper-node/pull/1968/files#r700620753

Reply:
Actually, only a single test used the `Sum` trait. I removed this and replaced the code with `try_fold` that allows `checked_add` to be used

Reply:
Addressed further in https://github.com/mpapierski/casper-node/commit/761b14992e257610481ca7c90da305ec55d67665 where I'm shadowing num_traits::Checked* ops with methods, and Add/Sub traits aren't implemented at all.